### PR TITLE
Add ClusterRole for the networkplugin_syncer

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

File: config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ resources:
   - role.yaml
   - role_binding.yaml
   - service_account.yaml
+  - networkplugin_syncer_cluster_role.yaml
+  - networkplugin_syncer_cluster_role_binding.yaml
```
Comment (tpantelis) [inline config/rbac/kustomization.yaml:13]: What is this file for?

Comment (skitt) [reply config/rbac/kustomization.yaml:13]: @SteveMattar you’ll probably be able to explain this better than me...

Comment (tpantelis) [reply config/rbac/kustomization.yaml:14]: I put these in here assuming they're needed.

Comment (SteveMattar) [reply config/rbac/kustomization.yaml:14]: @tpantelis its a list of rbac resources which `kustomize` is using to build the final bundle for OLM.
It's needed for operatorhub.io basically.

Comment (mangelajo) [reply config/rbac/kustomization.yaml:14]: I believe that if you run `kubectl kustomize .`  in this directory, you will get the bundle.yaml out (which is used to install the operator just from a yaml)

Comment (mangelajo) [reply config/rbac/kustomization.yaml:14]: but only the rbacs part

File: config/rbac/networkplugin_syncer_cluster_role.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+      - secrets
+      - nodes
+    verbs:
+      - get
+      - list
```
Comment (mangelajo) [inline config/rbac/networkplugin_syncer_cluster_role.yaml:16]: Can we remove - list for "secrets and configmaps", for nodes we need it.

We will limit a little bit the security this way.. at least the name of the secret/configmap needs to be know to be accessible.

File: pkg/subctl/operator/submarinerop/scc/ensure.go [hunk 3]
```diff
@@ -23,5 +23,11 @@ import (
 )
 
 func Ensure(restConfig *rest.Config, namespace string) (bool, error) {
-	return scc.UpateSCC(restConfig, namespace, serviceaccount.OperatorServiceAccount)
+	operatorUserAdded, err := scc.UpateSCC(restConfig, namespace, serviceaccount.OperatorServiceAccount)
+	if err != nil {
+		return false, err
+	}
+
+	npSyncerUserAdded, err := scc.UpateSCC(restConfig, namespace, serviceaccount.NPSyncerServiceAccount)
```
Comment (mangelajo) [inline pkg/subctl/operator/submarinerop/scc/ensure.go:31]: This one doesn't need to be on the SCC, it hopefully won't run as a privileged pod. (as far as I have identified)

File: pkg/subctl/operator/submarinerop/scc/ensure.go [hunk 4]
```diff
@@ -23,5 +23,11 @@ import (
 )
 
 func Ensure(restConfig *rest.Config, namespace string) (bool, error) {
-	return scc.UpateSCC(restConfig, namespace, serviceaccount.OperatorServiceAccount)
+	operatorUserAdded, err := scc.UpateSCC(restConfig, namespace, serviceaccount.OperatorServiceAccount)
```
Comment (mangelajo) [inline pkg/subctl/operator/submarinerop/scc/ensure.go:26]: unrelated to your patch, we have a typo in UpateSCC -> UpdateSCC

**Review by mangelajo — CHANGES_REQUESTED** [review]
Thank you @tpantelis just a couple of changes.

File: pkg/subctl/operator/submarinerop/scc/ensure.go (hunk 3 above)
Comment (tpantelis) [reply pkg/subctl/operator/submarinerop/scc/ensure.go:31]: ok - I wasn't sure what this stuff was for 😄

Comment (mangelajo) [reply pkg/subctl/operator/submarinerop/scc/ensure.go:31]: In openshift there are security contexts for service accounts, and for being allowed to escalate for use of hostnetwork pod, or privileged ports, you need to be on the privileged security context. ':)

File: pkg/subctl/operator/submarinerop/scc/ensure.go (hunk 4 above)
Comment (tpantelis) [reply pkg/subctl/operator/submarinerop/scc/ensure.go]: Fixed the typo

File: config/rbac/kustomization.yaml (hunk 1 above)
Comment (SteveMattar) [reply config/rbac/kustomization.yaml:14]: You can run `make bundle`

**Review by mangelajo — APPROVED** [review]

**Comment by mangelajo** [comment]
I have checked with : https://github.com/submariner-io/submariner/pull/943 and it's working as expected, the service account has the required permissions to work.

**Review by SteveMattar — COMMENTED** [review]
lgtm
Just one suggestion:
I would rename and add a prefix "is" to those variables inside every ensure function just to indicate they are type boolean
for example: `operatorSACreated` --> `isOperatorSACreated`

**Comment by tpantelis** [comment]
> lgtm
> Just one suggestion:
> I would rename and add a prefix "is" to those variables inside every ensure function just to indicate they are type boolean
> for example: `operatorSACreated` --> `isOperatorSACreated`

Or `createdOperatorSA` to be consistent with other parts of the code. I think that reads a little better.

**Review by SteveMattar — APPROVED** [review]

**Pull request merged**